src/coreComponents/mesh/utilities/MeshMapUtilities.hpp

```diff
@@ -0,0 +1,216 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2019 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2019 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2019 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All right reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file MeshMapUtilities.hpp
+ */
+#ifndef GEOSX_MESH_UTILITIES_MESHMAPUTILITIES_HPP
+#define GEOSX_MESH_UTILITIES_MESHMAPUTILITIES_HPP
+
+#include "common/DataTypes.hpp"
+
+namespace geosx
+{
+
+/**
+ * @brief This namespace contains helper functions that facilitate access
+ *        into the assortment of maps used by GEOSX mesh object managers
+ *        (e.g. array2d/array1d(array1d)/ArrayOfArrays/ArrayOfSets, etc.)
+ */
+namespace meshMapUtilities
+{
+
+//////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * @brief @return the size of the map along first dimension
+ * @tparam T type of map element
+ * @param map reference to the map
+ */
+template< typename T >
+inline localIndex size0( arrayView1d< arrayView1d< T const > const > const & map )
+{
+  return map.size();
+}
+
+/**
+ * @copydoc size0(arrayView1d< arrayView1d< T const > const > const &)
+ * @tparam USD unit-stride dimension of the map
+ */
+template< typename T, int USD >
+inline localIndex size0( arrayView2d< T, USD > const & map )
+{
+  return map.size( 0 );
+}
+
+/**
+ * @copydoc size0(arrayView1d< arrayView1d< T const > const > const &)
+ */
+template< typename T >
+inline localIndex size0( ArrayOfArraysView< T const > const & map )
+{
+  return map.size();
+}
+
+/**
+ * @copydoc size0(arrayView1d< arrayView1d< T const > const > const &)
+ */
+template< typename T >
+inline localIndex size0( ArrayOfSetsView< T const > const & map )
+{
+  return map.size();
+}
```

Comment:
Between `arrayView1d< arrayView1d>`, `ArrayOfSetsView` and `ArrayOfArraysView` for sure, but `arrayView2d` is different (`size()` with no arguments does not do what I want). I chose to explicitly spell out all supported types here rather than make a template function to handle 3 cases together (I didn't even need `T` parameter here tbh, since mesh maps are always `localIndex`). Or did you have something else in mind?